### DISCUSSION TITLE
Share data models between microservices

### DISCUSSION THREAD

**Discussion by kasperschnack** [pydantic/pydantic#4052] [Question]
I've tried to find opinions on this but haven't been successful so hopefully I can find one here. I find myself more and more often spending a lot of time keeping the data model in sync between an API and the application(s) that consumes it. To me it seems that it would be better to import the model from somewhere separate to avoid having to keep the data model in sync two or more places.

1) is this not a good idea? if so why not?
2) if it is a good idea, then do you guys have any nifty ways of achieving this?

**Comment by FossenWang** [comment]
I'm used to package a SDK in the API project, and install the SDK where needs to access the API server.
So, the data models are in the SDK package, shared with API server and applications.

**Reply by kasperschnack** [reply]
How do you package it?

**Reply by FossenWang** [reply]
`python setup.py bdist_wheel`
Then install the whl package directly or upload to private pypi.
